src/ui/App.js

```diff
@@ -1,43 +1,135 @@
 // @flow
 
 import React from "react";
-import {Admin, Resource, ListGuesser, EditGuesser} from "react-admin";
+import {Admin, Resource, ListGuesser, EditGuesser, Loading} from "react-admin";
+import {
+  InitiativeList,
+  InitiativeCreate,
+  InitiativeEdit,
+} from "./components/Initiatives";
 import fakeDataProvider from "ra-data-fakerest";
 import {createMemoryHistory} from "history";
+import {createMuiTheme} from "@material-ui/core/styles";
+import {Graph, type Node} from "../core/graph";
+import {fakeInitiatives} from "./mock/fakeInitiatives";
 
-async function loadAndReport(path) {
+async function loadAndReport(path: string) {
   const response = await fetch(path);
   if (!response.ok) {
     console.error(path, response);
   }
-  const json = await response.json();
-  console.log(path, json);
+  return response.json();
 }
 
-const dataProvider = fakeDataProvider({
-  people: [
-    {id: 0, title: "Hello, world!"},
-    {id: 1, title: "FooBar"},
-    {id: 2, title: "Bar foo"},
-  ],
-  comments: [
-    {id: 0, person_id: 0, author: "John Doe", body: "Sensational!"},
-    {id: 1, person_id: 0, author: "Jane Doe", body: "I agree"},
-  ],
-});
+function loadUsers(plugins, graph): Array<Object> {
+  const userPrefixes = plugins.reduce((acc, {userTypes}) => {
+    userTypes.forEach((type) => acc.push(type.prefix));
+    return acc;
+  }, []);
+  const users = [];
+  userPrefixes.forEach((prefix) => {
+    const userIterator = graph.nodes({prefix});
+    let nextUser = userIterator.next();
+    while (!nextUser.done) {
+      const user = {
+        ...nextUser.value,
+      };
+      users.push(user); // find a way to return users, possibly refactor
+      nextUser = userIterator.next();
+    }
+  });
+  return users;
+}
+
+function loadActivities(plugins, graph): Array<Object> {
+  const activityPrefixes = plugins.reduce((acc, {nodeTypes}) => {
+    nodeTypes.forEach(
+      (type) =>
+        !["Bot", "User", "Identity", "Like"].find((n) => n === type.name) &&
+        acc.push(type.prefix)
+    );
+    return acc;
+  }, []);
+  const activities = [];
+  activityPrefixes.forEach((prefix) => {
+    const activityIterator = graph.nodes({prefix});
+    let nextActivity = activityIterator.next();
+    while (!nextActivity.done) {
+      const activity = {
+        ...nextActivity.value,
+      };
+      activities.push(activity); // find a way to return activities here, possibly refactor
+      nextActivity = activityIterator.next();
+    }
+  });
+  return activities;
+}
+
+const dataProvider = fakeDataProvider(fakeInitiatives, true);
 
 const history = createMemoryHistory();
 
-export default class App extends React.Component<{||}> {
+const theme = createMuiTheme({
+  palette: {
+    type: "dark",
+  },
+});
```

Comment:
Noticed that the text is hard to read in some of these fields, can we change the color to pink instead of blue easily?

![image](https://user-images.githubusercontent.com/7143583/85637539-b8a1e000-b640-11ea-93c7-9f272c6f27b8.png)